### ISSUE TITLE
Molnet train/valid/test logic

### ISSUE THREAD

**Issue by haozhenWu** [deepchem/deepchem#583]
Currently,  the performance report of DeepChem only has train/valid results. Is it true that molnet creates train/valid/test splits based on 0.8/0.1/0.1 ratio but only evaluate model performances on validation set?
That being said test data is always created but did not use ? Thanks.

**Comment by lilleswing** [comment]
@haozhenWu  are you discussing the code?  Or the moleculenet [paper](https://arxiv.org/pdf/1703.00564.pdf)?  The currently posted version of the paper on arxiv has test scores.

![selection_007](https://user-images.githubusercontent.com/1225605/26839695-cde1dab8-4ab1-11e7-866f-60df83ffa744.png)

If you want the code to print out test scores as well you simply have to pass the "test" flag to run_benchmark.
https://github.com/deepchem/deepchem/blob/master/deepchem/molnet/run_benchmark.py#L21

We still have to update the [readme](https://github.com/deepchem/deepchem) to include the latest MoleculeNet scores with test.  @miaecle.

**Comment by miaecle** [comment]
@haozhenWu Thanks for raising the issue, splitting is always 0.8/0.1/0.1 but we only evaluate test performances when needed. There is a flag in `dc.molnet.run_benchmark`: `test=False`, setting this to True allows you the retrieve the test set performance. Also the newer version of benchmark running script `benchmark2.py` in the example folder provides option `-t` to set the flag.

**Comment by haozhenWu** [comment]
@lilleswing Thanks for pointing me to the paper!
@miaecle Just to double check, the validation set is supposed to select the hyper-parameter and tune the model right? But in the benchmark run,  https://github.com/deepchem/deepchem/blob/master/deepchem/molnet/run_benchmark_models.py#L369 
it only pass train_dataset to build the model and evaluate on valid set. The validation set seems to be the same as test set in this case. Is it because the hyper-parameters are already selected based on the validation set ? Do I missed anything ? Thanks

**Comment by miaecle** [comment]
@haozhenWu Yes exactly, validation set and test set are regarded as the same in the benchmark run script. Basically we do all hyper-parameter search separately, hence it is not integrated into the script. Optimized hyper-parameters could be transported into the benchmark run by setting `hyper_parameters`  when calling the benchmark run function, or a default setting will be used.

**Comment by haozhenWu** [comment]
@miaecle That make sense. Thank you.
